test/utils/stringOrderField-test.ts

```diff
@@ -0,0 +1,344 @@
+
+import { midPointsBetweenStrings, reorderLexicographically } from "../../src/utils/stringOrderField";
+
+const moveLexicographicallyTest = (
+    orders: Array<string | undefined>,
+    fromIndex: number,
+    toIndex: number,
+    expectedChanges: number,
+    maxLength?: number,
+): void => {
+    const ops = reorderLexicographically(orders, fromIndex, toIndex, maxLength);
+
+    const zipped: Array<[number, string | undefined]> = orders.map((o, i) => [i, o]);
+    ops.forEach(({ index, order }) => {
+        zipped[index][1] = order;
+    });
+
+    const newOrders = sortBy(zipped, i => i[1]);
+    expect(newOrders[toIndex][0]).toBe(fromIndex);
+    expect(ops).toHaveLength(expectedChanges);
+};
+
+describe("stringOrderField", () => {
+    it("stringToBase", () => {
+        expect(Number(stringToBase(""))).toBe(0);
+        expect(Number(stringToBase(" "))).toBe(1);
+        expect(Number(stringToBase("a"))).toBe(66);
+        expect(Number(stringToBase(" !"))).toBe(97);
+        expect(Number(stringToBase("aa"))).toBe(6336);
+        expect(Number(stringToBase("cat"))).toBe(620055);
+        expect(Number(stringToBase("doggo"))).toBe(5689339845);
+        expect(Number(stringToBase("a", "abcdefghijklmnopqrstuvwxyz"))).toEqual(1);
+        expect(Number(stringToBase("a"))).toEqual(66);
+        expect(Number(stringToBase("c", "abcdefghijklmnopqrstuvwxyz"))).toEqual(3);
+        expect(Number(stringToBase("ab"))).toEqual(6337);
+        expect(Number(stringToBase("cb", "abcdefghijklmnopqrstuvwxyz"))).toEqual(80);
+        expect(Number(stringToBase("AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"))).toEqual(4.648312045971824e+78);
+        expect(Number(stringToBase("~".repeat(50)))).toEqual(7.776353884348688e+98);
+        expect(Number(stringToBase("      "))).toEqual(7820126496);
+        expect(Number(stringToBase("  "))).toEqual(96);
+        expect(Number(stringToBase(" !"))).toEqual(97);
+        expect(Number(stringToBase("S:J\\~"))).toEqual(4258975590);
+        expect(Number(stringToBase("!'Tu:}"))).toEqual(16173443434);
+    });
+
+    it("baseToString", () => {
+        expect(baseToString(BigInt(10))).toBe(DEFAULT_ALPHABET[9]);
+        expect(baseToString(BigInt(10), "abcdefghijklmnopqrstuvwxyz")).toEqual("j");
+        expect(baseToString(BigInt(6241))).toEqual("`a");
+        expect(baseToString(BigInt(53), "abcdefghijklmnopqrstuvwxyz")).toEqual("ba");
+        expect(baseToString(BigInt(1234))).toBe("+}");
+        expect(baseToString(BigInt(0))).toBe("");
+        expect(baseToString(BigInt(1))).toBe(" ");
+        expect(baseToString(BigInt(95))).toBe("~");
+        expect(baseToString(BigInt(96))).toBe("  ");
+        expect(baseToString(BigInt(97))).toBe(" !");
+        expect(baseToString(BigInt(98))).toBe(' "');
+        expect(baseToString(BigInt(1))).toBe(" ");
+    });
+
+    it("midPointsBetweenStrings", () => {
```

Comment:
conventions say that `it` should follow with a `should <expectation>`, otherwise you're `describe`ing

```
describe('functionName', () => {
  it('should work', () => {});
});
```

Reply:
applies throughout